lib/spack/spack/environment/environment.py

```diff
@@ -1414,6 +1416,8 @@ def _spec_needs_overwrite(self, spec):
         return mtime > record.installation_time
 
     def _get_overwrite_specs(self):
+        # TODO: Can the cache even be populated at this point?
+        self._spec_needs_overwrite.cache.clear()
```

Comment:
I think the only way this could be populated at this point is if the user is using `spack-python` and doing multiple spack installs from the same python shell.